client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -187,7 +187,9 @@ private void failConnection(Throwable throwable) {
                     log.warn("Connection for segment {} failed due to: {}", segmentName, throwable.getMessage());
                 }
             }
-            if (failSetupConnection) {
+            if (throwable instanceof SegmentSealedException || throwable instanceof NoSuchSegmentException) {
+                setupConnection.releaseExceptionally(throwable);
```

Comment:
SSS drops the connection when the client tries to write on the stream (that was deleted and recreated). ` io.pravega.client.segment.impl.SegmentOutputStreamImpl$ResponseProcessor.connectionDropped`. 
Should we handle this scenario too?

Reply:
That is ok. It's normal for connection dropped to come in late. You can see several checks related to that above like on 183, and 186. I don't see anything wrong with the current handling of that case.